### PR TITLE
Remove JAXB annotation processor from Jersey JSON provider

### PR DESCRIPTION
**Pull request by clicktravel-steffan**
Signed-off-by: Steffan Westcott <steffan.westcott@clicktravel.com>

### CODE REVIEW

File: commons/commons-json-provider/src/main/java/com/clicktravel/common/http/application/ObjectMapperProvider.java [hunk 1]
```diff
@@ -37,13 +33,12 @@
...
-                new JaxbAnnotationIntrospector(TypeFactory.defaultInstance())));
         objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         objectMapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
         objectMapper.setSerializationInclusion(Include.NON_NULL);
 
-        // TODO Uncomment following code when completing migration of JSON provider to Jackson
+        // Swagger codegen generates the following code, but it seems to be redundant
+        // as the Java model files for enums have @JsonValue on the toString() methods
         // objectMapper.enable(SerializationFeature.WRITE_ENUMS_USING_TO_STRING);
         // objectMapper.enable(DeserializationFeature.READ_ENUMS_USING_TO_STRING);
```
Comment (clicktravel-chrishern) [inline commons/commons-json-provider/src/main/java/com/clicktravel/common/http/application/ObjectMapperProvider.java]: Could we remove this commented out code?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Mon Sep 09 23:13:46 BST 2013
+#Tue Jul 19 00:15:11 BST 2016
```
Comment (clicktravel-chrishern) [inline gradle/wrapper/gradle-wrapper.properties]: Just wondering if this file was intentionally committed as it doesn't appear to be related to the changes in hand?

**Comment by clicktravel-chrishern** [comment]
@clicktravel-steffan Review done - just a couple of questions.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (clicktravel-steffan) [inline gradle/wrapper/gradle-wrapper.properties]: This is an incidental tweak, as suggested by IntelliJ

File: commons/commons-json-provider/src/main/java/com/clicktravel/common/http/application/ObjectMapperProvider.java (hunk 1 above)
Comment (clicktravel-steffan) [inline commons/commons-json-provider/src/main/java/com/clicktravel/common/http/application/ObjectMapperProvider.java]: I wanted to retain this information, as it may help us in future when we move to using Swagger codegen for generating clients

**Comment by clicktravel-chrishern** [comment]
@clicktravel-steffan Ok - happy with both of those.  Feel free to merge when you're ready.
